### PR TITLE
Return error if default exists

### PR DESCRIPTION
**Pull request by wg102**
Related to https://jirab.statcan.ca/browse/BTIS-367

### CODE REVIEW

**Comment by wg102** [comment]
This was pushed to das-dev and works as expected
image

The only way to have the vutton is to already be on homepage, without a default notebook, when you edit an existing notebook to add the default tag

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -27,11 +28,12 @@ jobs:
           - 5000:5000
     steps:
     # Determine if pushing to ACR or DEV ACR
+    # Bypassed for dev, only push if has auto-deploy tag
     - name: Set ENV variables for a PR
-      if: github.event_name == 'pull_request'
-      run: echo "REGISTRY=${{env.DEV_REGISTRY_NAME}}.azurecr.io" >> "$GITHUB_ENV"
+      if: github.event_name == 'pull_request' && contains( github.event.pull_request.labels.*.name, 'auto-deploy')
```
Comment (mathis-marcotte) [inline .github/workflows/publish.yml:33]: not needed in the if statement

File: notebooks.go [hunk 2]
```diff
@@ -499,6 +499,20 @@ func (s *server) createDefaultNotebook(namespace string, notebookNames []string,
 	var notebook newnotebookrequest
 	notebookname := namespace + "-notebook"
 
+	notebookRequirement, err := labels.NewRequirement("notebook.statcan.gc.ca/default-notebook", selection.Exists, []string{})
+	if err != nil {
+		return notebook, errors.New("an error occurent checking for a default notebook")
+	}
+	labelSelector := labels.NewSelector().Add(*notebookRequirement)
+	pods, err := s.listers.notebooks.Notebooks(namespace).List(labelSelector)
```
Comment (mathis-marcotte) [inline notebooks.go:507]: rename "pods" to "notebooks" for more accurate name

**Review by mathis-marcotte — APPROVED** [review]
lgtm

**Pull request merged**